CONTRIBUTING.md

```diff
@@ -0,0 +1,120 @@
+
+*   **Reporting Bugs**: If you find a crash, wrong result, or build failure, please file a [Bug Report](.github/ISSUE_TEMPLATE/01_bug_report.yml).
+*   **Performance Tuning**: Bolt is all about speed. If you identify a bottleneck, file a [Performance Issue](.github/ISSUE_TEMPLATE/06_performance_issue.yml) or submit an optimization PR.
+*   **Documentation**: Improve our docs, fix typos, or translate guides.
+*   **Answering Questions**: Help other users in [Discussions](https://github.com/bytedance/bolt/discussions).
+
+## 🛠 Development Setup
+
+Bolt uses modern C++ tooling. Please ensure your environment meets the requirements.
+
+### Prerequisites
+*   **OS**: Linux (Ubuntu 20.04+, CentOS 7+). macOS is currently experimental.
+*   **Compiler**: GCC 10+ or Clang 12+ (C++17 support required).
+*   **Build Tools**: CMake 3.20+, Ninja.
+*   **Dependency Manager**: Conan 2.
+
+### Build Instructions
+
+```bash
+# 1. Clone the repository
+git clone https://github.com/bytedance/bolt.git
+cd bolt
+
+# 2. build for Presto/Spark
+make release # for Presto
+make release_spark  # for Spark
+```
+
+> **Note**: For development and debugging, you might want to use `make debug` However, strictly use `make release` for performance benchmarking.
+
+## 🎨 Code Style & Conventions
+
+We strictly enforce code style to maintain a consistent codebase.
+
+*   **C++ Standard**: We use **C++17**.
+*   **Formatting**: We use `clang-format-14`. Please run the formatter before submitting code:
+    ```bash
+    # Run check
+    make clang-format-check
+    
+    # Apply fix for all files
+    make clang-format
+    # Apply fix for modified files
+    make clang-format-modified
+    ```
+*   **Naming**: Follow the [Style Guide](CODING_STYLE.md).
+*   **Modern C++**: Prefer smart pointers standard algorithms over raw pointers and manual loops.
+
+## 🧪 Testing & Verification
+
+Code without tests is technical debt.
+
+### Running Unit Tests
+All PRs must pass the unit test suite.
+```bash
+cd bolt
+make unittest_release
+make unittest_release_spark
+```
+
+### Sanitizers (Crucial for C++)
+Before submitting a complex PR, it is highly recommended to run tests with AddressSanitizer (ASAN) and UndefinedBehaviorSanitizer (UBSAN) to catch memory leaks and undefined behaviors.
+
+```bash
+cd bolt
+ENABLE_ASAN=True make unittest_release
+```
+
+## 🚀 Performance & Benchmarking
+
+Since Bolt is a compute engine, **performance regressions are treated as bugs**.
+
+*   **Micro-benchmarks**: If you optimize a specific function (e.g., `substr`), please add or run the corresponding google-benchmark.
+*   **Reporting**: In your PR, you **Strongly recommended** provide a "Before vs. After" performance comparison if your change affects the critical path.
+
+## 📥 Submitting a Pull Request
+
+1.  **Fork & Branch**: Create a new branch for your feature (e.g., `feature/array-sort` or `fix/memory-leak`).
+2.  **Commit Messages**: Write clear, descriptive commit messages.
```

Comment:
I think we should consider adopting conventional commits: https://www.conventionalcommits.org/en/v1.0.0/